tutorials/haskell/snap-web-app/tutorial.md

```diff
@@ -0,0 +1,482 @@
+[SQLite](https://www.sqlite.org/), and [Stack](http://docs.haskellstack.org/en/stable/README/)
+as our Haskell build tool. The target app should deliver the folllowing features:
+
+ - List Todo items
+ - Create a new Todo item
+ - Mark a Todo item as Done
+ - Delete a Todo item
+
+
+## Getting Started
+
+Snap comes shipped with a [scaffolding utility](http://snapframework.com/docs/quickstart)
+to scrap some boilerplate. So, when you first run the `snap init` command inside an empty
+directory, it will create a bunch of files for you. But let's leave that aside,
+for now let's focus on how to create a Snap web app from scratch using Stack.
+
+Firstly, let's [create a project using Stack](http://docs.haskellstack.org/en/stable/README/#start-your-new-project) by running the following command:
+
+```
+$ stack new todo-webapp
+```
+
+Then you can edit the cabal file to match the one provided in this tutorial. There's a couple of things you won't
+understand, but don't worry about it. The important thing is the
+`build-depends` section of your library which should look like:
+
+```
+build-depends:         base                      >= 4.7     && < 5
+                     , bytestring                >= 0.9.1   && < 0.11
+                     , heist                     >= 0.14    && < 0.15
+                     , lens                      >= 4.12    && < 4.14
+                     , mtl                       >= 2       && < 3
+                     , snap                      >= 0.13    && < 0.15
+                     , snap-core                 >= 0.9     && < 0.10
+                     , snap-extras               >= 0.11    && < 0.12
+                     , snap-server               >= 0.9     && < 0.10
+                     , snap-loader-static        >= 0.9     && < 0.10
+                     , text                      >= 0.11    && < 1.3
+                     , xmlhtml                   >= 0.1     && < 0.3
+                     , HDBC                      >= 2.3     && <= 2.5
+                     , HDBC-sqlite3              == 2.3.*
+                     , interpolate               >= 0.1 && < 0.2
+```
+
+Those are the required libraries for the application to be built. Most of them
+belongs to Snap's core like [`snap`](http://hackage.haskell.org/package/snap), [`snap-core`](http://hackage.haskell.org/package/snap-core) ,
+[`snap-server`](http://hackage.haskell.org/package/snap-server)
+and [`heist`](http://hackage.haskell.org/package/heist). Been this last the preferred template library by Snap's creators. There are
+others libraries like [`HDBC`](http://hackage.haskell.org/package/HDBC) and
+[`HDBC-sqlite3`](http://hackage.haskell.org/package/HDBC-sqlite3), which enable
+interaction with our database, and some others like [`text`](http://hackage.haskell.org/package/text)
+and [`interpolate`](http://hackage.haskell.org/package/interpolate) to
+allow us inmutable unicode text manipulation and strings interpolation.
```

Comment:
@nebtrx 

"Most of them **belong** (instead of belongs) to Snap's core**:** `snap`, `snap-core`, `snap-server`**,** and `heist`."

For the "Been this last..." sentence, maybe parentheses or a more complete sentence works:

"... `heist` (the template library preferred by the Snap creators)"

Or something else to not begin the sentence with "Been this last." Also, "HTML templating library" is probably clearer for describing `heist`.

"`text` and `interpolate` **for** **immutable** **U**nicode text manipulation..."